src/main/java/org/cryptomator/ui/mainwindow/ResizeController.java

```diff
@@ -119,6 +89,21 @@ private void checkDisplayBounds(WindowEvent evt) {
 		}
 	}
 
+	private boolean isOutOfDisplayBounds() {
+		final int x = settings.windowXPosition.get();
+		final int y = settings.windowYPosition.get();
+		final int width = settings.windowWidth.get();
+		final int height = settings.windowHeight.get();
+		return isRectangleOutOfScreen(x + 20, y, 0, height) // Left pixel column (+20)
+				|| isRectangleOutOfScreen(x + width - 20, y, 0, height) // Right pixel column (-20)
+				|| isRectangleOutOfScreen(x, y + 5, width, 0) // Top pixel row (+5)
+				|| isRectangleOutOfScreen(x, y + height - 20, width, 0); // Bottom pixel row (-20)
+	}
+
+	private boolean isRectangleOutOfScreen(int x, int y, int width, int height) {
+		return Screen.getScreensForRectangle(x, y, width, height).isEmpty();
+	}
+
```

Comment:
Ok, I suggest using the window's position instead of the saved position because they may not be the same, resulting in doing the check on old saved values. Also, the rows and columns that we check should not cross each other otherwise the edge case will only reset when the entire window is completely out of bounds.